analysis_driver/report_generation/run_crawler.py

```diff
@@ -103,7 +103,7 @@ def _populate_barcode_info_from_dataset(self, dataset):
         for project_id in self.projects:
             self.projects[project_id][ELEMENT_SAMPLES] = list(self.projects[project_id][ELEMENT_SAMPLES])
 
-        if dataset.has_barcodes:
+        if dataset.has_barcode_in_lane(run_element[ELEMENT_LANE]):
```

Comment:
Here we will process only the last run element iterated over in `dataset.run_elements` - is this really what we want?

Reply:
Very good catch. That would have caused mixed flowcell to not have the unassigned run element if the last lane was not a pool